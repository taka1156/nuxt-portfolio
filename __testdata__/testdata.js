const MAX_NAV_DATA = 4;
const MAX_PROFILE_DATA = 4;
const MAX_CARD_DATA = 6;
const dummyImgUrl = 'http://placehold.jp/150x150.png';
const GITHUB_URL = 'https://github.com/taka1156/audio-player';
const WEBSITE_URL = 'https://takatechblog.netlify.app/';

// factory
const dummyFactory = (num, fn) => {
  return [...new Array(num)].map((_, i) => fn(i + 1));
};

// template
const dummyNavTemplate = () => ({
  name: 'ダミーナビゲーション',
  to: '/test1',
  img: dummyImgUrl,
});

const dummyCardTemplate = () => ({
  title: 'ダミーカード',
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
});

const dummyCardLinkTemplate = i => ({
  title: 'ダミーカード',
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  link: i % 2 === 0 ? GITHUB_URL : WEBSITE_URL,
});

const dummyProfileTemplate = () => {
  return 'ダミープロフ';
};

const dummyImg = {
  imgUrl: dummyImgUrl,
  imgAlt: 'ダミーの画像',
};

const dummyRoutes = dummyFactory(MAX_NAV_DATA, () => dummyNavTemplate());

const dummyCards = dummyFactory(MAX_CARD_DATA, () => dummyCardTemplate());

const dummyCardsLink = dummyFactory(MAX_CARD_DATA, i => dummyCardLinkTemplate(i));

const dummyProfiles = dummyFactory(MAX_PROFILE_DATA, () => dummyProfileTemplate());

const dummyLogo = 'ダミーロゴ';

export {
  dummyImg,
  dummyLogo,
  dummyRoutes,
  dummyCards,
  dummyCardsLink,
  dummyProfiles,
};
