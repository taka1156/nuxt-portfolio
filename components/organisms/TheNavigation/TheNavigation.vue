<template>
  <div>
    <nav>
      <nav-bar :logo-text="logoText" :is-open="isOpen" @change-state="changeState" />
      <nav-list :is-open="isOpen" :routes="routes" @change-state="changeState" />
    </nav>
  </div>
</template>

<script>
import NavBar from '../../molecules/NavBar/NavBar';
import NavList from '../../molecules/NavList/NavList';

export default {
  name: 'TheNavigation',
  components: {
    NavBar,
    NavList
  },
  props: {
    /**
     * ロゴテキスト
     */
    logoText: {
      type: String,
      required: true
    },
    /**
     * ナビゲーションの項目が定義された配列
     */
    routes: {
      type: Array,
      required: true
    }
  },
  data() {
    return {
      isOpen: false
    };
  },
  methods: {
    changeState(isOpen) {
      /**
       * ナビゲーションの開閉状態を変更
       */
      this.isOpen = isOpen;
    }
  }
};
</script>
