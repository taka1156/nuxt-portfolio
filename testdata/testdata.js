const dummyLogo = 'ダミー';

const dummyRoutes = [
  {
    name: 'nav-item',
    to: '/test1',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test2',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test3',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test4',
    img: 'http://placehold.jp/150x150.png',
  },
];

const dummyProfile = [
  'TypeScriptの学習',
  'Laravel+Vueでサービスを作る',
  'AWSなどインフラ周りの学習',
  'CSS/SASSの学習',
];

const dummyCards = [
  {
    title: 'PHP',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '1~4ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Laravel',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '2~3ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'JavaScript(ES6)',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '６ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Vue.js',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '8ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
];

const dummyCardsLink = [
  {
    title: 'ブログサイト',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Nuxt.js(Vue.js), axios, microCMS, BootStrap4, vue-infinite-loading',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://takatechblog.netlify.app/',
  },
  {
    title: 'マークダウンエディター',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, BootStrap4',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/markdown-editor',
  },
  {
    title: 'LineBot',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Express.js, axios, @line/bot-sdk , Vercel(Zeit Now)',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/line-bot',
  },
  {
    title: 'オーディオプレイヤー',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, JSMediaTags',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/audio-player',
  },
];

export { dummyLogo, dummyRoutes, dummyProfile, dummyCards, dummyCardsLink };
