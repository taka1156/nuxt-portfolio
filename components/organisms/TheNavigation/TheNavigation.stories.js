import TheNavigation from './TheNavigation.vue';
import { dummyLogo, dummyRoutes } from '@/__testdata__/testdata.js';

export default {
  title: 'Organisms/TheNavigation',
  component: TheNavigation
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { TheNavigation },
  template: '<the-navigation v-bind="$props" />'
});

export const Default = Template.bind({});
Default.args = { logoText: dummyLogo, routes: dummyRoutes };
