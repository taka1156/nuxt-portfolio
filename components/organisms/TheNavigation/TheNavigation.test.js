import { mount, RouterLinkStub } from '@vue/test-utils';
import TheNavigation from './TheNavigation.vue';
import { dummyLogo, dummyRoutes } from '@/__testdata__/testdata.js';

describe('TheNavigation', () => {
  const theNavigation = mount(TheNavigation, {
    stubs: {
      NuxtLink: RouterLinkStub,
    },
    propsData: {
      logoText: dummyLogo,
      routes: dummyRoutes,
    },
  });

  it('NavBar初期値: logoText, routes', () => {
    // logoText
    expect(theNavigation.vm.$options.props.logoText.required).toBe(true);
    expect(theNavigation.vm.logoText).toBe(dummyLogo);
    // routes
    expect(theNavigation.vm.$options.props.routes.required).toBe(true);
    console.log(theNavigation.vm.$options.props.routes.default());
    expect(theNavigation.vm.routes).toBe(dummyRoutes);
  });

  it('閉じた時のスナップショット', () => {
    expect(theNavigation.html()).toMatchSnapshot();
  });

  it('changeStateのテスト', () => {
    expect(theNavigation.vm.$data.isOpen).toBe(false);
    theNavigation.vm.changeState(true);
    expect(theNavigation.vm.$data.isOpen).toBe(true);
  });

  it('開いた時のスナップショット', () => {
    expect(theNavigation.html()).toMatchSnapshot();
  });
});
