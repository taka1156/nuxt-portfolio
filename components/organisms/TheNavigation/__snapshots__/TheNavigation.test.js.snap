// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNavigation 閉じた時のスナップショット 1`] = `
"<div>
  <nav>
    <div>
      <div class=\\"nav-bar\\">
        <div class=\\"nav-bar__box\\"><a class=\\"base-link--extend\\">ダミー</a>
          <div class=\\"base-nav-icon base-nav-icon--extend\\"><button><span class=\\"base-nav-icon__border base-nav-icon__top base-nav-icon__top--close\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__middle\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__bottom base-nav-icon__bottom--close\\"></span> <span class=\\"base-nav-icon__text\\">
        NAVI
      </span></button></div>
        </div>
      </div>
    </div>
    <div>
      <transition-stub name=\\"nav-fade\\">
        <div class=\\"nav-list\\" style=\\"display: none;\\">
          <ul>
            <li>
              <div>
                <div class=\\"nav-item\\"><a class=\\"base-link--extend\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミー1の画像\\" class=\\"base-img--extend base-img--lg\\">
                    <p class=\\"base-text--extend\\">ダミー1</p>
                  </a></div>
              </div>
            </li>
            <li>
              <div>
                <div class=\\"nav-item\\"><a class=\\"base-link--extend\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミー2の画像\\" class=\\"base-img--extend base-img--lg\\">
                    <p class=\\"base-text--extend\\">ダミー2</p>
                  </a></div>
              </div>
            </li>
            <li>
              <div>
                <div class=\\"nav-item\\"><a class=\\"base-link--extend\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミー3の画像\\" class=\\"base-img--extend base-img--lg\\">
                    <p class=\\"base-text--extend\\">ダミー3</p>
                  </a></div>
              </div>
            </li>
            <li>
              <div>
                <div class=\\"nav-item\\"><a class=\\"base-link--extend\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミー4の画像\\" class=\\"base-img--extend base-img--lg\\">
                    <p class=\\"base-text--extend\\">ダミー4</p>
                  </a></div>
              </div>
            </li>
          </ul>
        </div>
      </transition-stub>
    </div>
  </nav>
</div>"
`;

exports[`TheNavigation 開いた時のスナップショット 1`] = `
"<div>
  <nav>
    <div>
      <div class=\\"nav-bar\\">
        <div class=\\"nav-bar__box\\"><a class=\\"base-link--extend\\">ダミー</a>
          <div class=\\"base-nav-icon base-nav-icon--extend\\"><button><span class=\\"base-nav-icon__border base-nav-icon__top base-nav-icon__top--open\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__middle base-nav-icon__middle--fade\\"></span> <span class=\\"base-nav-icon__border base-nav-icon__bottom base-nav-icon__bottom--open\\"></span> <span class=\\"base-nav-icon__text\\">
        CLOSE
      </span></button></div>
        </div>
      </div>
    </div>
    <div>
      <transition-stub name=\\"nav-fade\\">
        <div class=\\"nav-list\\" style=\\"\\">
          <ul>
            <li>
              <div>
                <div class=\\"nav-item\\"><a class=\\"base-link--extend\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミー1の画像\\" class=\\"base-img--extend base-img--lg\\">
                    <p class=\\"base-text--extend\\">ダミー1</p>
                  </a></div>
              </div>
            </li>
            <li>
              <div>
                <div class=\\"nav-item\\"><a class=\\"base-link--extend\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミー2の画像\\" class=\\"base-img--extend base-img--lg\\">
                    <p class=\\"base-text--extend\\">ダミー2</p>
                  </a></div>
              </div>
            </li>
            <li>
              <div>
                <div class=\\"nav-item\\"><a class=\\"base-link--extend\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミー3の画像\\" class=\\"base-img--extend base-img--lg\\">
                    <p class=\\"base-text--extend\\">ダミー3</p>
                  </a></div>
              </div>
            </li>
            <li>
              <div>
                <div class=\\"nav-item\\"><a class=\\"base-link--extend\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミー4の画像\\" class=\\"base-img--extend base-img--lg\\">
                    <p class=\\"base-text--extend\\">ダミー4</p>
                  </a></div>
              </div>
            </li>
          </ul>
        </div>
      </transition-stub>
    </div>
  </nav>
</div>"
`;
